test/tests/functional/pbs_multi_sched.py

```diff
@@ -1637,3 +1637,103 @@ def test_dedicated_time_multisched(self):
         self.server.expect(JOB, {ATTR_comment: "Not Running: " + msg}, id=jid2)
         self.scheds['sc2'].log_match(jid2 + ";Dedicated Time",
                                      starttime=self.server.ctime)
+
+    def test_auto_sched_off_due_to_fds_limit(self):
+        """
+        Test to make sure scheduling should be turned off automatically
+        when number of open files per process are exhausted
+        """
+
+        # get the number of open files per process
+        ret = self.du.run_cmd(
+            self.server.hostname, [
+                'ulimit', '-n'], as_script=True, logerr=False)
+        self.assertEqual(ret['rc'], 0)
+        open_files_limit = ret['out'][0]
+
+        # set the number of open files per process to 10
+        ret = self.du.run_cmd(
+            self.server.hostname, [
+                'ulimit', '-n', '10'], as_script=True, logerr=False)
+        self.assertEqual(ret['rc'], 0)
```

Comment:
You should use resource.setrlimit()

Reply:
Adding to what Bhroam mentioned, You will have to do this change a superuser.

Reply:
Done